python/tchannel/tornado/connection.py

```diff
@@ -593,19 +605,39 @@ def send_request(self, request):
         # TODO: fire before_receive_response
 
         def adapt_tracing(f):
+            if f.exception():
+                # TODO unexpected exception
+                response_future.set_exception(
+                    f.exception()
+                )
+                return
+        
             # fetch the request tracing for response
             f.result().tracing = request.tracing
-            response_future.set_result(f.result())
-            # event: receive_response
-            if self.tchannel:
-                self.tchannel.event_emitter.fire(
-                    EventType.after_receive_response,
-                    f.result(),
+
+            if isinstance(f.result(), ProtocolError):
+                protocol_error = f.result()
+                response_future.set_exception(
+                    TChannelException(protocol_error.message)
                 )
+                # event: after_receive_protocol_error
+                if self.tchannel:
```

Comment:
We should kill all of these `if self.tchannel:` checks if they're only for some legacy test code -- we can just update the test code.